mailbox/api/src/main/java/org/apache/james/mailbox/model/MailboxAnnotationKey.java

```diff
@@ -91,6 +95,18 @@ public String asString() {
         return key.toLowerCase();
     }
 
+    public boolean isInclusiveParentOf(MailboxAnnotationKey key) {
+        int thatComponentsSize = key.countComponents();
+        int thisComponentsSize = this.countComponents();
+        return (thatComponentsSize == thisComponentsSize || thatComponentsSize == thisComponentsSize + 1)
+            && key.asString().startsWith(this.asString());
+    }
+
+    public boolean isInclusiveAncestorOf(MailboxAnnotationKey key) {
+        return key.countComponents() >= this.countComponents()
+            && key.asString().startsWith(this.asString());
```

Comment:
I don't want to duplicate this in implementation, as it should not have much knowledge about what's an AnnotationKey. I don't like doing complicated, hardly readable DB queries for this, when we can have easily readable, well named code.

However, you are right, you spotted a bug! Thanks ;-)